### DISCUSSION TITLE
Conditional link styling

### DISCUSSION THREAD

**Discussion by fabianoliver** [Blazor-Diagrams/Blazor.Diagrams#348] [Q&A]
Hi,

first of all - amazing library, I love the thought that went into customizability.

I'm using this to display data flow in a distributed system. I would like to style links based on properties of the link itself.  Picture something like "if there's currently data in transmit between two nodes, display the link line with some nice animation" (and similarly for other possible states of the link)

Do we currently have any reasonably straightforward way to do that?

So far, I'm being a bit cheap and just changing the link's Color property, but obviously it would be nice to be able to apply other styles here. Conceptually, I'd imagine a nice way might be creating a custom LinkModel with some custom property, then having the renderer apply classes based on the property value. But so far the only way I've seen would be to implement a custom link widget for that, which seems a touch excessive.

Maybe there's some nicer way to achieve that already? Or if not, would that be something that's easy to add in some standardised way to the library?

**Comment by zHaytam** [comment]
Hello,

Thank you for the kind words!

To be honest, what you're describing is exactly the use of a custom link model and widget.  
For the widget, you can just copy the code for the library one and add logic to it.

I can't add these specific scenarios to the library because there will always be more things to add.

**Reply by zHaytam** [reply]
You can also try something like this in your custom link model:

```cs
public new string Color
{
// custom get/set logic
}
```

But to be honestly it's kind of ugly, it would be better to have both custom model and widget so that in the future if you want to add or customize something else, it's not a pain.

**Reply by fabianoliver** [reply]
Thanks, no worries - that makes sense.

For now, an easier workaround came to mind: I can assign each state to an individual color, even if just ever so slightly different, and then map a css selector in the likeness of path[stroke="that_particular_color"] to add more properties. 

Yes, it's massively ugly, but its so quick and easy that I might just go down that route for now :) If time permits, I might look at a proper custom link model further down the road though.
